### PR TITLE
peertracker package logging

### PR DESCRIPTION
**Pull request by amoore877**
**Pull Request check list**

- [X] Commit conforms to CONTRIBUTING.md?
- [X] Proper tests/regressions included?
- [NA] Documentation updated?

**Affected functionality**
`peertracker` package

**Description of change**
Plumb logging through trackers/watchers.
Add relevant fields to logs, logging generally alongside the more generic returned errors from functions.

**Which issue this PR fixes**
#2412

### CODE REVIEW

File: pkg/common/peertracker/tracker_linux.go [hunk 1]
```diff
@@ -106,10 +132,14 @@ func (l *linuxWatcher) IsAlive() error {
 	// TODO: Evaluate the use of `starttime` as the primary exit detection mechanism.
 	currentStarttime, err := getStarttime(l.pid)
 	if err != nil {
-		return fmt.Errorf("caller exit suspected due to failure to get starttime: %w", err)
+		msg := "caller exit suspected due to failure to get starttime"
+		l.log.WithError(err).Warn(msg)
+		return errors.New(msg)
```
Comment (rturner3) [inline pkg/common/peertracker/tracker_linux.go:137]: Can we retain the error from the call to `getStarttime()` that was previously wrapped in the new error we pass back to the caller?

File: pkg/common/peertracker/tracker_bsd.go [hunk 2]
```diff
@@ -195,7 +208,9 @@ func (b *bsdWatcher) IsAlive() error {
 
 	select {
 	case <-b.done:
-		return errors.New("caller exit detected via kevent notification")
+		msg := "caller exit detected via kevent notification"
+		b.log.Warn(msg)
```
[the remark this answers is not in the document]
Comment (amoore877) [reply pkg/common/peertracker/tracker_bsd.go:212]: [Using my own standards against me I see](https://github.com/spiffe/spire/commit/d09e61720a98b03d445920d0e1863d9dae4e87dd) :) the nerve

Was focused on migration and forgot the opportunity to do cleanups. Will do.

File: pkg/common/peertracker/tracker_linux.go [hunk 3]
```diff
@@ -90,10 +112,14 @@ func (l *linuxWatcher) IsAlive() error {
...
 	if err != nil {
-		return fmt.Errorf("caller exit suspected due to failed readdirent: err=%w", err)
+		msg := "caller exit suspected due to failed readdirent"
+		l.log.WithError(err).Warn(msg)
+		return errors.New(msg)
 	}
 	if n < 0 {
-		return fmt.Errorf("caller exit suspected due to failed readdirent: n=%d", n)
+		msg := fmt.Sprintf("caller exit suspected due to failed readdirent: n=%d", n)
+		l.log.Warn(msg)
```
[the remark this answers is not in the document]
Comment (amoore877) [reply pkg/common/peertracker/tracker_linux.go]: done

File: pkg/common/peertracker/tracker_bsd.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (amoore877) [reply pkg/common/peertracker/tracker_bsd.go]: done

File: pkg/common/peertracker/tracker_linux.go (hunk 1 above)
Comment (amoore877) [reply pkg/common/peertracker/tracker_linux.go]: sure. done

**Review by rturner3 — APPROVED** [review]
LGTM

File: pkg/common/peertracker/tracker_bsd.go [hunk 4]
```diff
@@ -9,6 +9,13 @@ import (
 	"sync"
 	"syscall"
 	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/spire/pkg/common/telemetry"
+)
+
+const (
+	_bsdType = "bsd"
```
Comment (azdagron) [inline pkg/common/peertracker/tracker_bsd.go:18]: nit: this naming does not follow the conventions for unexported package level consts in the codebase
```suggestion
	bsdType = "bsd"
```

File: pkg/common/peertracker/tracker_linux.go [hunk 5]
```diff
@@ -81,6 +102,7 @@ func (l *linuxWatcher) IsAlive() error {
 	defer l.mtx.Unlock()
 
 	if l.procfd < 0 {
+		l.log.Warn("Caller is no longer being watched")
```
Comment (azdagron) [inline pkg/common/peertracker/tracker_linux.go:105]: I'm a little torn here. While in general I think creating structured fields is a good idea, I can't help but notice that this logging could also be accomplished with a simple refactor:

```
func (l *linuxWatcher) IsAlive() error {
	if err := l.isAlive(); err != nil {
		l.log.WithError(err).Warn("Peertracker liveness check failed")
		return err
	}
	return nil
}

func (l *linuxWatcher) isAlive() error {
	// ... current IsAlive impl ...
}
```

**Review by azdagron — COMMENTED** [review]
Looks ok to me. I have one food-for-thought comment...

File: pkg/common/peertracker/tracker_linux.go (hunk 5 above)
Comment (amoore877) [reply pkg/common/peertracker/tracker_linux.go:105]: True, though we'd lose the searchable old vs. new value fields, which could make it harder to organize or analyze logging data without first needing to do string/regex parsing.

**Review by azdagron — APPROVED** [review]

File: pkg/common/peertracker/tracker_linux.go (hunk 5 above)
Comment (azdagron) [reply pkg/common/peertracker/tracker_linux.go:105]: Fair enough!

File: pkg/common/peertracker/tracker_linux.go [hunk 6]
```diff
@@ -9,19 +9,30 @@ import (
 	"strings"
 	"sync"
 	"syscall"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/spire/pkg/common/telemetry"
+)
+
+const (
+	_linuxType = "linux"
```
Comment (azdagron) [inline pkg/common/peertracker/tracker_linux.go:18]: Sorry, i should have pointed this one out as well.

Comment (amoore877) [reply pkg/common/peertracker/tracker_linux.go:18]: I should have been more attentive to seeing other places to follow through :)

**Comment by amoore877** [comment]
It's not obvious to me from CI why one particular test failed:
```
Removing network spire-integration-7gueoa_our-network
Network spire-integration-7gueoa_our-network not found.
[2021-08-24T17:43:09Z] cleaning up services...
Removing network spire-integration-7gueoa_our-network
Network spire-integration-7gueoa_our-network not found.
[2021-08-24T17:43:10Z] "upgrade" test suite succeeded.
[2021-08-24T17:43:10Z] The following tests failed: nested-rotation
Error: Process completed with exit code 123.
```

**Review by azdagron — APPROVED** [review]

**Pull request merged**